tests/acceptance/01_vars/02_functions/classfilterdata_object_of_objects_endogenous.cf

```diff
@@ -0,0 +1,73 @@
+body common control
+{
+  bundlesequence => { "test", "check" };
+}
+
+bundle agent test
+{
+  meta:
+    "description" -> { "CFE-4562" }
+      string => "Test for expected results from classfilterdata() with object of objects using endogenous key";
```

Comment:
replace endogenous key with something less technical, as I suggested "self_key" for "exogenous_key", maybe this would be "internal_key"?